rpc/src/module/test.rs

```diff
@@ -165,4 +159,71 @@ impl IntegrationTestRpc for IntegrationTestRpcImpl {
             Ok(hash.unpack())
         }
     }
+
+    fn generate_block_with_template(&self, block_template: BlockTemplate) -> Result<H256> {
+        let snapshot: &Snapshot = &self.shared.snapshot();
+        let consensus = snapshot.consensus();
+        let parent_header = snapshot
+            .get_block_header(&block_template.parent_hash.pack())
+            .expect("parent header should be stored");
+        let mut seen_inputs = HashSet::new();
+
+        let txs: Vec<_> = block_template
+            .transactions
+            .iter()
+            .map(|tx_template| {
+                let transaction: packed::Transaction = tx_template.data.clone().into();
+                transaction.into_view()
+            })
+            .collect();
+
+        let transactions_provider = TransactionsProvider::new(txs.as_slice().iter());
+        let overlay_cell_provider = OverlayCellProvider::new(&transactions_provider, snapshot);
+
+        let rtxs: Vec<_> = txs.iter().filter_map(|tx| {
+            resolve_transaction(
+                tx.clone(),
+                &mut seen_inputs,
+                &overlay_cell_provider,
+                snapshot,
+            ).map_err(|err| {
```

Comment:
The snapshot's alive cellset is based on `snapshot.tip` but not `parent`.
So I doubt this line has a bug.
For example, the `tx` spends an input cell that is alive for `parent` but not for `snapshot.tip`.

Reply:
unresolvable tx will be ignored by `filter_map`, we cannot find a way to provide parent block's live cell set in current code.

Reply:
I see that the current code cannot provide the alive cellset for non-tip blocks.
Then I suggest that do not implicitly filter the unresolvable transactions but just explicitly throw an error to describe why this RPC cannot accept these transactions. As a user, a filtered block is not my expected.

Reply:
updated, please review again, thanks.